### DISCUSSION TITLE
Compatibility with PG-Strom

### DISCUSSION THREAD

**Discussion by JoshInnis** [orioledb/orioledb#370] [General]
Hello,

I have a project I would like to work on that involves many Postgres' extensions either working together, or at least not interfering with each other.

Part of this project involves using OrioleDB and another extension called [PG-Strom](https://github.com/heterodb/pg-strom). I have already confirmed that I cannot expect to use PG-Strom on tables that does not use the [heap TAM](https://github.com/heterodb/pg-strom/discussions/823).

However, now I am looking into whether is possible these two extensions to work on the same database cluster at the same time. Cost prohibits me from using more than one machine. Both this extension and that extension use the Shared Memory hooks and the Object Access Hooks.

Is there any known incompatibilities between these two systems that I should be aware of, before I try to get them working together on my own?

**Comment by AntheaLiles** [comment]
Hi, 

just want to follow up,

@JoshInnis btw which extensions are you currently using ?
I'm designing a data fabric for building management and I plan to use Timescale for IoTs, PostGIS for (well...) GIS data and Apache AGE for all kind of networking things

In future development, we are looking to further extend our solution with AI oriented feature such as pgVectors and the PostgresML framework.

I'm currently working on scalability opportunity, 
I've been looking different solutions : pg-stom, citus, pg_partman, pgCat...

OrioleDB could be put on the table for several uses cases too

But i'm wondering if all of this tooling can keep working together seemlessly... The stack complexity kinda explode these days
